[vulkan] mm op through addmm

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #41435 WIP - vulkan view
* #41434 [vulkan] cat op (concatenate)
* #41380 [vulkan] inplace add_, relu_
* #41379 [vulkan] max_pool2d
* #41224 [vulkan][asan] Fix Invalid Memory ops
* #41223 [vulkan] reshape op
* #41222 [vulkan] support add for dim < 4
* **#41221 [vulkan] mm op through addmm**

Differential Revision: [D22754938](https://our.internmc.facebook.com/intern/diff/D22754938)

IvanKobzarev committed:
[vulkan] mm op through addmm

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"




[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"




[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"




[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"


Differential Revision: [D22754938](https://our.internmc.facebook.com/intern/diff/D22754938)

[ghstack-poisoned]

AshkanAliabadi commented on aten/src/ATen/native/vulkan/VulkanAten.cpp:
```diff
@@ -294,6 +294,25 @@ Tensor vulkan_addmm(
+      self.dim() == 2 && mat2.dim() == 2,
+      "vulkan_mm expects 2-dimensional tensors");
+  auto m1Sizes = self.sizes();
+  auto m2Sizes = mat2.sizes();
```
const please.  Here and anywhere else applicable.

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"


Differential Revision: [D22754938](https://our.internmc.facebook.com/intern/diff/D22754938)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"


Differential Revision: [D22754938](https://our.internmc.facebook.com/intern/diff/D22754938)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[vulkan] mm op through addmm"


Differential Revision: [D22754938](https://our.internmc.facebook.com/intern/diff/D22754938)

[ghstack-poisoned]
